### PR TITLE
Publish packages from CircleCI

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR add a publish step to `package` jobs. It pushes packages to a dedicated S3 bucket. Also, it introduces building packages for rockylinux, almalinux and ubuntu.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/175614) / small_tests / b970db10da0bbd0f9b827028d9f776678d06d2e7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4026/175614/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4026/175614/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-18_09.42.50/lib.mongooseim.logs/run.2023-05-18_09.42.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/176189) / small_tests / 3f0b0aa4c73586ee28f2b26ac36b493b96621f91
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4026/176189/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4026/176189/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-23_10.38.49/lib.mongooseim.logs/run.2023-05-23_10.38.49/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/176729) / small_tests / 501f74900a7f5136ca1f82b1720f8f3f91afdef4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4026/176729/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4026/176729/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-29_09.45.44/lib.mongooseim.logs/run.2023-05-29_09.45.44/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/176923) / small_tests / a37492ee7ff2f82e04dba8908c93258bfdd21f2f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4026/176923/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4026/176923/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-31_11.17.53/lib.mongooseim.logs/run.2023-05-31_11.17.54/suite.log.html)

File: .circleci/template.yml [hunk 1]
```diff
@@ -748,14 +752,49 @@ workflows:
...
+          platform: rockylinux_8
+          context: mongooseim-org
+          otp_package: "24.1.7-1"
+          filters: *all_tags
+      - package:
+          name: almalinux_8
+          executor: otp_24
+          platform: almalinux_8
+          context: mongooseim-org
+          otp_package: "24.1.7-1"
```
Comment (arcusfelis) [inline .circleci/template.yml:769]: any reason to use an older version?

File: tools/pkg/publish.sh [hunk 2]
```diff
@@ -1,41 +1,21 @@
...
 fi
 
-set +e
-PKG=$(ls -1 ${PACKAGES_LOCAL} | grep "$VERSION-$REVISION" | head -1)
-if [ -z "$PKG" ]; then
-    echo "$0: can't find package version $VERSION-$REVISION"
-    exit 2
-fi
-set -e
+aws configure set aws_access_key_id $PKG_ACCESS_KEY_ID --profile $PKG_PROFILE
```
Comment (arcusfelis) [inline tools/pkg/publish.sh:18]: we usually wrap variables into `""`.

File: tools/pkg/publish.sh [hunk 3]
```diff
@@ -1,41 +1,21 @@
...
-
-if [ x"$(echo $CHKSUM | cut -d" " -f1)" != x"$(echo $REMOTE_CHKSUM | cut -d" " -f1)" ]; then
-    echo "$0: checksum mismatch - try again or check your connection"
-    exit 3
-fi
-ssh $PACKAGES_HOST "sed -i /$VERSION-$REVISION/d $PACKAGES_PATH/sha256sum " \
-                   "&& echo $REMOTE_CHKSUM >> $PACKAGES_PATH/sha256sum"
-echo Package $PKG is now public at:
-echo https://$PACKAGES_HOST/$PACKAGES_PROJECT/$PKG
+aws s3 cp ${PACKAGE_NAME} s3://arn:aws:s3:${PKG_AWS_REGION}:767271724337:accesspoint/mim-packages/$prefix/${PACKAGE_NAME} --acl public-read --profile $PKG_PROFILE --region $PKG_AWS_REGION
```
Comment (arcusfelis) [inline tools/pkg/publish.sh:21]: 767271724337 could be an env variable too

**Review by arcusfelis — COMMENTED** [review]
added some comments :)

File: .circleci/template.yml (hunk 1 above)
Comment (pawlooss1) [reply .circleci/template.yml:769]: Currently there is no otp 25 package available for almalinux and rockylinux. I believe that when the new site with packages is running we could expect some newer version published.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/177018) / small_tests / 9c366e5836ce3d74754e79994c1f014418d41c5a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4026/177018/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4026/177018/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-06-02_10.15.12/lib.mongooseim.logs/run.2023-06-02_10.15.13/suite.log.html)

**Comment by pawlooss1** [comment]
Marking as a draft because we need to discuss naming convention.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/177479) / small_tests / 5c2ed1f22a4b1fc91870083ed0b24060be681d17
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4026/177479/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4026/177479/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-06-13_08.28.13/lib.mongooseim.logs/run.2023-06-13_08.28.13/suite.log.html)

**Review by arcusfelis — APPROVED** [review]
looks good

**Pull request merged**